generator/generator.go

```diff
@@ -0,0 +1,138 @@
+	}
+	err = helpers.GenerateMachineJson(resourceData) // generate schema/base/machine.json.d/machine-generated.json
+	if err != nil {
+		return fmt.Errorf("error generating machine json err=%v", err)
+	}
+	return nil
+}
+
+func loadBlacklist() []string {
+	blacklist := os.Getenv("RANCHER_DRIVER_BLACKLIST")
+	if "" != blacklist {
+		return strings.Split(blacklist, ",")
+	}
+	// defaulting to this list for now
+	return []string{}
+}
+
+func fetchData() (*helpers.ResourceData, error) {
+	blacklist := loadBlacklist()
+	resourceMap := make(map[string]helpers.ResourceFields)
+	for _, driver := range drivers.GetDriverNames() {
+		resourceFieldStruct := make(helpers.ResourceFields)
+		resourceFieldMap := make(helpers.ResourceFieldConfigs)
+		resourceFieldStruct["resourceFields"] = resourceFieldMap
+		resourceMap[driver] = resourceFieldStruct
+		createFlags, err := drivers.GetCreateFlagsForDriver(driver)
+		if err != nil {
+			return nil, fmt.Errorf("error getting create flags for driver=%s, err=%v", driver, err)
+		}
+		for _, flagStruct := range createFlags {
+			var flagName, flagType string
+			var err error
+			switch flagStruct.(type) {
+			case cli.StringFlag:
+				flag := flagStruct.(cli.StringFlag)
+				flagName, err = getRancherName(flag.Name)
+				if err != nil {
+					return nil, fmt.Errorf("error getting the rancher name of flagStruct=%v for driver=%s, err=%v", flagStruct, driver, err)
+				}
+				flagType = "string"
+			case cli.IntFlag:
+				flag := flagStruct.(cli.IntFlag)
+				flagName, err = getRancherName(flag.Name)
+				if err != nil {
+					return nil, fmt.Errorf("error getting the rancher name of flagStruct=%v for driver=%s, err=%v", flagStruct, driver, err)
+				}
+				flagType = "string"
+			case cli.BoolFlag:
+				flag := flagStruct.(cli.BoolFlag)
+				flagName, err = getRancherName(flag.Name)
+				if err != nil {
+					return nil, fmt.Errorf("error getting the rancher name of flagStruct=%v for driver=%s, err=%v", flagStruct, driver, err)
+				}
+				flagType = "boolean"
+			case cli.BoolTFlag:
+				flag := flagStruct.(cli.BoolTFlag)
+				flagName, err = getRancherName(flag.Name)
+				if err != nil {
+					return nil, fmt.Errorf("error getting the rancher name of flagStruct=%v for driver=%s, err=%v", flagStruct, driver, err)
+				}
+				flagType = "boolean"
+			default:
+				return nil, fmt.Errorf("unknown type of flag %v, for driver=%s", flagStruct, driver)
+			}
+			resourceFieldMap[flagName] = helpers.ResourceFieldConfig{Type: flagType, Nullable: true, Required: false}
+		}
+	}
+	return &helpers.ResourceData{Blacklist: blacklist, Drivers: drivers.GetDriverNames(), ResourceMap: resourceMap}, nil
+}
+
+func getRancherName(machineFlagName string) (string, error) {
+	parts := strings.SplitN(machineFlagName, "-", 2)
+	if len(parts) != 2 {
+		errString := fmt.Sprintf("parameter %s does not follow expected naming convention [DRIVER]-[FLAG-NAME]", machineFlagName)
+		return "", fmt.Errorf(errString)
```

Comment:
No need to fmt.Sprintf and fmt.Errorf. just put the string and its parameters into Errorf